lib/node-package-manager.ts

```diff
@@ -48,9 +24,41 @@ export class NodePackageManager implements INodePackageManager {
 			}
 
 			try {
-				return this.loadAndExecute("install", [pathToSave, packageName], { config: config }).wait();
+				let jsonContentBefore = this.$fs.readJson(path.join(pathToSave, "package.json")).wait();
+				let dependenciesBefore: Array<string> = [];
+				if(jsonContentBefore && jsonContentBefore.dependencies) {
+					for(let name in jsonContentBefore.dependencies) {
+						dependenciesBefore.push(name);
+					}
+				}
+
+				let flags = this.getFlagsString(config, true);
+				let params = ["install", packageName];
+				params = params.concat(flags);
+				this.$childProcess.spawnFromEvent("npm", params, "close", { cwd: pathToSave }).wait();
+
+				let jsonContentAfter = this.$fs.readJson(path.join(pathToSave, "package.json")).wait();
+				let dependenciesAfter: Array<string> = [];
+				if(jsonContentAfter && jsonContentAfter.dependencies) {
+					for(let name in jsonContentAfter.dependencies) {
+						dependenciesAfter.push(name);
+					}
+				}
+
+				/** This diff is done in case the installed pakcage is a URL address, a path to local directory or a .tgz file
+				 *  in these cases we don't have the package name and we can't rely on "npm install --json"" option
+				 *  to get the project name because we have to parse the output from the stdout and we have no controll over it (so other messages may be mangled in stdout)
+				 * 	The solution is to traverse node_modules before and after install and get the name of the installed package,
+				 * 	even if it's installed through local path or URL. If command installes more than one package, all package names are returned.
+				 */
+				let diff = dependenciesAfter.filter(x => dependenciesBefore.indexOf(x) === -1);
```

Comment:
I've found a case where this approach will not work. Consider the case where a plugin is already installed in package.json, but user wants to update it's version (or install it from .tgz this time). In this case the difference will be empty array and the plugin will not be installed. I would suggest to compare both the name of the plugin and the version. For example this code could do the job:
```TypeScript
let diff = _(jsonContentAfter.dependencies)
    .omitBy((val: string, key: string) => jsonContentBefore && jsonContentBefore.dependencies[key] && jsonContentBefore.dependencies[key] === val)
    .keys()
    .value();
```